packages/firestore/type-test.ts

```diff
@@ -147,7 +165,7 @@ getDocsFromServer(
 ).then();
 onSnapshot(doc(firebase.firestore(), 'foo', 'foo'), () => {});
 onSnapshot(doc(firebase.firestore(), 'foo', 'foo'), {
-  next: (snapshot: FirebaseFirestoreTypes.DocumentSnapshot) => {
```

Comment:
I'm a little concerned about the typing items being breaking changes - what are your thoughts on that? Will this be drop in for people when this feature ships, or will some typescript code need updating.

We can issue breaking changes, that isn't a problem. The problem is accidentally issuing a non-major semantic release version with breaking changes - we cannot do that.

Reply:
If the types are awkward but nevertheless follow firebase-js-sdk then no problem - awkward still but we have  "taste" reference that is "just follow firebase-js-sdk". If they are awkward but divergent - and especially if we are doing a breaking change for types, then this would be a good time to true them up.

I'll reiterate that I'm not afraid of a breaking change (we're on semver major number 23 after all) I just try to be as empathetic as possible and include clear and succinct "this change, impact is that, likely adaptation is ..." so API users can take advantage of what we already know instead of spending limited daily thinking time having to derive it themselves

So the goal then - if this is a breaking change - is just to do it really well, make good docs, then send it...